nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -204,6 +189,39 @@ impl DataStore {
         .await
     }
 
+    /// List all IP Pools, making as many queries as needed to get them all
+    ///
+    /// This should generally not be used in API handlers or other
+    /// latency-sensitive contexts, but it can make sense in saga actions or
+    /// background tasks.
+    pub async fn ip_pools_list_batched(
+        &self,
+        opctx: &OpContext,
+        reservation_type: IpPoolReservationType,
+        version: Option<IpVersion>,
+    ) -> ListResultVec<IpPool> {
+        opctx.check_complex_operations_allowed()?;
+        let mut pools = Vec::new();
+        let mut paginator = Paginator::new(
+            SQL_BATCH_SIZE,
+            dropshot::PaginationOrder::Ascending,
+        );
+        while let Some(p) = paginator.next() {
```

Comment:
Would it be simpler and more efficient to just do a direct query without pagination here instead of iterating through pages when we just want everything in one go?

Reply:
Probably. I think we've done this in a bunch of places where latency isn't that important, or at least, less important than potentially putting pressure on the database. If it's alright with you, I'd like to keep it as-is, since it really isn't latency-sensitive. Does that sound OK?